.github/workflows/release.yml

```diff
@@ -233,3 +235,53 @@ jobs:
           gh pr create --title "$message" \
                        --body "$body" \
                        --base main
+
+  update-apidiff-baseline-to-released-version:
+    runs-on: ubuntu-latest
+    needs:
+      - release
+    steps:
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          ref: main
+
+      - name: Use CLA approved github bot
+        run: .github/scripts/use-cla-approved-github-bot.sh
+
+      - name: Wait for release to be available in maven central
+        run: |
+          until curl --silent \
+                     --show-error \
+                     --output /dev/null \
+                     --head \
+                     --fail \
+                     https://repo1.maven.org/maven2/io/opentelemetry/javaagent/opentelemetry-javaagent/$VERSION/opentelemetry-javaagent-$VERSION.jar
+          do
+            sleep 60
```

Comment:
There is a timeout for github actions so the workflow would get killed after 6h (i think) guess that should be enough. An alternative would be to use something similar to cloudfoundry index updates that are run periodically and query the maven metadata for the latest version. Though would need to figure out how to get the previous version then.

Reply:
yeah, the artifacts have been showing up pretty quickly in maven central lately, though I haven't been paying super close attention, it will be interesting to see exactly how long this step runs